packages/here-wallet/src/lib/here-wallet.ts

```diff
@@ -0,0 +1,141 @@
+import type {
+  WalletModuleFactory,
+  WalletBehaviourFactory,
+  BrowserWallet,
+} from "@near-wallet-selector/core";
+import { createAction } from "@near-wallet-selector/wallet-utils";
+import * as BN from "bn.js";
+import {
+  getHereBalance,
+  HereConfiguration,
+  hereConfigurations,
+  setupWalletState,
+  transformTransactions,
+} from "./utils";
+
+export type HereWallet = BrowserWallet & {
+  getHereBalance: () => Promise<BN>;
+  getAvailableBalance: () => Promise<BN>;
+};
```

Comment:
These are unnecessary methods but that make it possible to get the correct wallet balance.
Here Wallet users often transfer all their near tokens into a here wallet smart contract for liquid staking. Therefore, the classic way to get a balance does not return the full amount of tokens available for spending. getAvailableBalance solve this problem and these methods are well documented in the readme. It seems to me that each wallet has the right to add some of its own custom methods if this will help library users work more efficiently, am I wrong?

```ts
const isHereWallet = (w: Wallet): w is HereWallet =>
  w.id === "here-wallet";
  
if (isHereWallet(wallet)) {
  wallet.getAvailableBalance(); // correct typings
}
```